integration/simple-cache-client.test.ts

```diff
@@ -60,87 +67,151 @@ describe('SimpleCacheClient.ts Integration Tests', () => {
     await withCache(momento, cacheName, async () => {
       await momento.set(cacheName, 'key', 'value');
       const res = await momento.get(cacheName, 'key');
-      expect(res.text()).toEqual('value');
+      expect(res instanceof CacheGet.Error).toEqual(false);
+      if (res instanceof CacheGet.Hit) {
+        expect(res.valueString()).toEqual('value');
+      }
     });
   });
-  it('should throw CacheNotFoundError if deleting a non-existent cache', async () => {
+  it('should return NotFoundError if deleting a non-existent cache', async () => {
     const cacheName = v4();
     const momento = new SimpleCacheClient(AUTH_TOKEN, 1111);
-    await expect(momento.deleteCache(cacheName)).rejects.toThrow(NotFoundError);
+    const deleteResponse = await momento.deleteCache(cacheName);
+    expect(deleteResponse instanceof DeleteCache.Error).toEqual(true);
+    if (deleteResponse instanceof DeleteCache.Error) {
+      expect(deleteResponse.errorCode()).toEqual(
+        MomentoErrorCode.NOT_FOUND_ERROR
+      );
```

Comment:
```suggestion
    expect(deleteResponse instanceof DeleteCache.Error).toEqual(true);
    expect(deleteResponse.errorCode()).toEqual(
      MomentoErrorCode.NOT_FOUND_ERROR
    );
```

Reply:
same here.

Reply:
If I take the second `expect` out of the `if`, I lose the pattern matching/casting to the subclass. I could add an `else` clause to the `if` statement instead, of using a redundant `instanceof` call, but I don't think that really buys me anything.